### ISSUE TITLE
/Wall + /WX break build on MSVC when using NMake

### ISSUE THREAD

**Issue by stolyaroleh** [PixarAnimationStudios/OpenSubdiv#953]
CMakeLists.txt has the following, starting from line 214:
```
elseif(MSVC)

    # Turn on all warnings
    list(APPEND OSD_COMPILER_FLAGS /Wall)

    list(APPEND OSD_COMPILER_FLAGS
                    /W3     # Use warning level recommended for production purposes.
                    /WX     # Treat all compiler warnings as errors.
```

The code adds `/Wall`, `/W3` and `/WX` to a list of compiler flags. A Visual Studio project, generated by CMake, picks `/W3`, but NMake seems to be using `/Wall`, and this breaks build on the first [info-level](https://stackoverflow.com/questions/4001736/whats-up-with-the-thousands-of-warnings-in-standard-headers-in-msvc-wall) warning.

I'd like to submit a pull request to fix this. What would be the best way to proceed?

1. Just remove `/Wall`, leaving `/W3`.
2. Remove `/WX` too, since it's not future proof in a sense that a new compiler version can introduce new warnings that will prevent builds.

Additionally, I think it's a good idea to simplify lines 128-132 to remove CMake warnings regarding policy [0054](https://cmake.org/cmake/help/latest/policy/CMP0054.html):
```
# Detect Clang (until a cmake version provides built-in variables)
if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
    set(CMAKE_COMPILER_IS_CLANGCC 1)
elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
    set(CMAKE_COMPILER_IS_ICC 1)
endif()
```
The syntax is `if(<variable|string> MATCHES regex)`, which means that we can use variable name instead of its value.

**Comment by davidgyu** [comment]
Thanks for these suggestions! Guidelines for contributing are here:
  https://github.com/PixarAnimationStudios/OpenSubdiv/blob/master/documentation/contributing.rst

The next step would be to return a signed Contributor License Agreement (CLA) so that we can accept pull requests from you.

-David

**Comment by stolyaroleh** [comment]
Just sent a signed CLA.

**Comment by jtran56** [comment]
Filed as internal issue #153338.

**Comment by davidgyu** [comment]
Thanks again for these notes! Fixed by #1318 and #1180
